subsys/usb/host/usbh_device.c

```diff
@@ -475,6 +477,58 @@ bool usbh_device_is_root(struct usbh_context *const ctx,
 	return sys_dlist_peek_head(&ctx->udevs) == &udev->node;
 }
 
+struct usb_device *usbh_connect_device(struct usbh_context *const ctx,
+					    uint8_t speed)
+{
+	struct usb_device *udev;
+
+	LOG_DBG("Device connected event");
+
+	/* Allocate new device */
+	udev = usbh_device_alloc(ctx);
+	if (udev == NULL) {
+		LOG_ERR("Failed allocate new device");
+		return NULL;
+	}
+
+	udev->state = USB_STATE_DEFAULT;
+	udev->speed = speed;
+
+	k_mutex_lock(&ctx->mutex, K_FOREVER);
+	sys_dlist_append(&ctx->udevs, &udev->node);
+	k_mutex_unlock(&ctx->mutex);
+
+	if (usbh_device_init(udev)) {
+		LOG_ERR("Failed to init new USB device");
+		k_mutex_lock(&ctx->mutex, K_FOREVER);
+		sys_dlist_remove(&udev->node);
+		k_mutex_unlock(&ctx->mutex);
```

Comment:
Likewise, `usbh_device_free()` takes care of calling `sys_dlis_remove();`, so not needed here (unless it is removed from `usbh_device_free()`.